checker.c

```diff
@@ -1,21 +1,157 @@
+ * if the SOC exceeds the 80% threshold, that reduces the life span of the battery, and losses are incurred
+ * Battery is charged above 80% only in outstation charging system.
+ * input: SOC in percentage
+ * returns: Check if the SOC is out of boundary conditions
+ *********************************************************************************/
+ 
+int BMS_StateOfCharge(float soc)
+{
+  int soc_check= ((soc < MINSOC) || (soc > MAXSOC));
+  if (soc_check)
+  {
+     printf("State of Charge is %f percent, and is out of range!\n", soc);
+     return 0;
   }
   return 1;
 }
 
+/********************************************************************************
+ * A function that gives Safe operating temperature during the charging of a Battery.
+ * There could be loss of charge if the temperature is beyond the boundary conditions
+ * input: Temperature in degrees
+ * returns: Check if the Temperature is out of boundary conditions
+ *********************************************************************************/
+ 
+int BMS_TemperatureCheck(float temperature_deg)
+{
+  int temperature_check= ((temperature_deg < MINTEMP) || (temperature_deg > MAXTEMP));
+  if(temperature_check)
+  {
+   printf("Temperature is %f and is out of range!\n", temperature_deg);
+    return 0;
+  } 
+  
+  printf(" The current BMS temperature is %f, and the conditions are ideal for charging the battery. \n", temperature_deg);
+  return 1;
+  
+}
+
+/********************************************************************************
+ * Process: Display the battery condition after all the factors are considered
+ *********************************************************************************/
+void BMS_DisplayCondition(int condition)
+{
+  if (condition)
+  {
+    printf(" The Battery management system is in good condition considering the above factors \n");
+  }
+  else
+  {
+   printf(" The Battery management system is in bad condition considering the above factors \n");
+  }
+}
+
+/********************************************************************************
+ * A function that gives overall status of a Battery management system
+ * Factors such as: Charging Temperature,Charge rate, SoH, SoC are considered to check if the BMS is good to function.
+ * input: Fators to check the plausibility of BMS
+ * returns: True is the factors meet the requirement
+ *********************************************************************************/
+ 
+int batteryIsOk(float StateofHealth, float ChargeRate, float stateofcharge, float temperature) 
+{
+  int socstatus, sohstatus, temperaturecheck, chargeratecheck,status;
+     sohstatus =  BMS_StateOfHealth(StateofHealth);
+     socstatus = BMS_StateOfCharge(stateofcharge);
+     chargeratecheck = BMS_ChargeRateCheck(ChargeRate);
+     temperaturecheck = BMS_TemperatureCheck(temperature);
+     status= (sohstatus & sohstatus & chargeratecheck & temperaturecheck);
```

Comment:
Spot the mistake here.
Write a test that will protect future maintainers from making such a mistake.